apollo-core/src/main/java/com/ctrip/framework/apollo/core/utils/DeferredLogger.java

```diff
@@ -0,0 +1,419 @@
+      DeferredLogCache.debug(logger, s, o1);
+    } else {
+      logger.debug(s, o, o1);
+    }
+  }
+
+  @Override
+  public void debug(String s, Object... objects) {
+    if (isEnabled()) {
+      DeferredLogCache.debug(logger, s, objects);
+    } else {
+      logger.debug(s, objects);
+    }
+  }
+
+  @Override
+  public void debug(String s, Throwable throwable) {
+    logger.debug(s, throwable);
+  }
+
+  @Override
+  public boolean isDebugEnabled(Marker marker) {
+    return logger.isDebugEnabled(marker);
+  }
+
+  @Override
+  public void debug(Marker marker, String s) {
+    logger.debug(marker, s);
+  }
+
+  @Override
+  public void debug(Marker marker, String s, Object o) {
+    logger.debug(marker, s, o);
+  }
+
+  @Override
+  public void debug(Marker marker, String s, Object o, Object o1) {
+    logger.debug(marker, s, o, o1);
+  }
+
+  @Override
+  public void debug(Marker marker, String s, Object... objects) {
+    logger.debug(marker, s, objects);
+  }
+
+  @Override
+  public void debug(Marker marker, String s, Throwable throwable) {
+    logger.debug(marker, s, throwable);
+  }
+
+  @Override
+  public boolean isInfoEnabled() {
+    return logger.isInfoEnabled();
+  }
+
+  @Override
+  public void info(String s) {
+    if (isEnabled()) {
+      DeferredLogCache.info(logger, s);
+    } else {
+      logger.info(s);
+    }
+  }
+
+  @Override
+  public void info(String s, Object o) {
+    if (isEnabled()) {
+      DeferredLogCache.info(logger, s, o);
+    } else {
+      logger.info(s, o);
+    }
+  }
+
+  @Override
+  public void info(String s, Object o, Object o1) {
+    if (isEnabled()) {
+      DeferredLogCache.info(logger, s, o, o1);
+    } else {
+      logger.info(s, o, o1);
+    }
+  }
+
+  @Override
+  public void info(String s, Object... objects) {
+    if (isEnabled()) {
+      DeferredLogCache.info(logger, s, objects);
+    } else {
+      logger.info(s, objects);
+    }
+  }
+
+  @Override
+  public void info(String s, Throwable throwable) {
+    logger.info(s, throwable);
+  }
+
+  @Override
+  public boolean isInfoEnabled(Marker marker) {
+    return logger.isInfoEnabled(marker);
+  }
+
+  @Override
+  public void info(Marker marker, String s) {
+    logger.info(marker, s);
+  }
+
+  @Override
+  public void info(Marker marker, String s, Object o) {
+    logger.info(marker, s, o);
+  }
+
+  @Override
+  public void info(Marker marker, String s, Object o, Object o1) {
+    logger.info(marker, s, o, o1);
+  }
+
+  @Override
+  public void info(Marker marker, String s, Object... objects) {
+    logger.info(marker, s, objects);
+  }
+
+  @Override
+  public void info(Marker marker, String s, Throwable throwable) {
+    logger.info(marker, s, throwable);
+  }
+
+  @Override
+  public boolean isWarnEnabled() {
+    return logger.isWarnEnabled();
+  }
+
+  @Override
+  public void warn(String s) {
+    if (isEnabled()) {
+      DeferredLogCache.warn(logger, s);
+    } else {
+      logger.warn(s);
+    }
```

Comment:
Is it necessary to implement all the methods? maybe by calling the overloaded methods?

```suggestion
    this.warn(s, (Object) null);
```

Reply:
Basic coverage common methods